pkgs/stdenv/darwin/default.nix

```diff
@@ -167,19 +167,50 @@ in rec {
 
   stage0 = stageFun 0 null {
     overrides = self: super: with stage0; {
-      coreutils = { name = "bootstrap-stage0-coreutils"; outPath = bootstrapTools; };
-      gnugrep   = { name = "bootstrap-stage0-gnugrep";   outPath = bootstrapTools; };
+      coreutils = stdenv.mkDerivation {
+        name = "bootstrap-stage0-coreutils";
+        buildCommand = ''
+          mkdir -p $out
+          ln -s ${bootstrapTools}/bin $out/bin
+        '';
+      };
+
+      gnugrep = stdenv.mkDerivation {
+        name = "bootstrap-stage0-gnugrep";
+        buildCommand = ''
+          mkdir -p $out
+          ln -s ${bootstrapTools}/bin $out/bin
+        '';
+      };
 
       darwin = super.darwin // {
         Libsystem = stdenv.mkDerivation {
           name = "bootstrap-stage0-Libsystem";
           buildCommand = ''
             mkdir -p $out
-            ln -s ${bootstrapTools}/lib $out/lib
+
+            cp -r ${self.darwin.darwin-stubs}/usr/lib $out/lib
+            chmod -R +w $out/lib
+            substituteInPlace $out/lib/libSystem.B.tbd --replace /usr/lib/system $out/lib/system
+
+            ln -s libSystem.B.tbd $out/lib/libSystem.tbd
+
+            for name in c dbm dl info m mx poll proc pthread rpcsvc util gcc_s.10.4 gcc_s.10.5; do
+              ln -s libSystem.tbd $out/lib/lib$name.tbd
```

Comment:
Hmm, why do we need this? I would expect all of these to be part of the stubs.

Reply:
The current implementation of stub generation collects the contents of system libraries, but not their aliases. This same structure is also present in the [main definition of libSystem](https://github.com/nixos/nixpkgs/blob/9d92c9d0a797b19662968b250919d075317d7379/pkgs/os-specific/darwin/apple-source-releases/Libsystem/default.nix#L100-L103).

Reply:
Right, so why don't we add these symlinks in the stubs repo? That way it's in one place.

Reply:
I've opened a PR to include the symlinks in the stubs repo: https://github.com/NixOS/darwin-stubs/pull/5. The stubs repo is a stub of (a slice of) an entire system, so there's still some work required to figure out which symlinks belong to libSystem and which are unrelated. I'm imagining something like

```sh
while read -r alias; do
  target=$(readlink "$alias")
  if [ "$target" == "libSystem.tbd" ] || [ "$target" == "libSystem.B.tbd" ]; then
    cp -d "$alias" "$out/usr/lib"
  fi
done < <(echo ${darwin-stubs}/usr/lib/*.tbd)
```

Reply:
Ah I see, was hoping this wouldn't need any logic in nixpkgs anymore. But regardless this is an improvement over the hardcoded list.

Perhaps the Libsystem build could be reworked a bit so the same expression can be used for both the final and bootstrap variants.

Reply:
I tried implementing this but ran into a snag with libutil. Current `darwin.Libsystem` defines this as a symlink alias to `libSystem.tbd`, but in the source version of macOS, this is a distinct library. Now that the aliases are more precise, there is no "libutil" alias so `libuv` fails due to linking with `-lutil`. The easy solution is to add `libutil` to the list of stubbed libraries and include this in the definition of `darwin.Libsystem`, but I'm wondering if I should stub all the libraries instead.

See comparisons for
 - [stubbing libutil](https://github.com/NixOS/darwin-stubs/compare/main...thefloweringash:stub-libutil?expand=1)
 - [stubbing all libraries](https://github.com/NixOS/darwin-stubs/compare/main...thefloweringash:stub-all-libs?expand=1)